src/main-process/parse-command-line.js

```diff
@@ -134,6 +138,18 @@ module.exports = function parseCommandLine (processArgs) {
     devMode = true
   }
 
+  let projectSettings = {}
+  if (atomProject) {
+    const contents = Object.assign({}, readProjectSettingsSync(atomProject, executedFrom))
+    const config = contents.config
+    const originPath = atomProject
+    const paths = contents.paths.map((curPath) =>
+      relativizeToAtomProject(curPath, path.dirname(path.join(executedFrom, atomProject))
+    ))
+    pathsToOpen.push(path.dirname(atomProject))
```

Comment:
what's this for?

Reply:
Atom's logic expects a path to be sent in the pathsToOpen array. I pass this in so it knows it needs to open paths, and then call replace before it loads.